ledger/participant-state/kvutils/src/test/suite/scala/com/daml/ledger/participant/state/kvutils/committer/TransactionCommitterSpec.scala

```diff
@@ -98,32 +135,60 @@ class TransactionCommitterSpec extends WordSpec with Matchers with MockitoSugar
     }
   }
 
-  "validateLedgerTime" should {
-    "continue without accessing ledger configuration if record time is not available" in {
-      val instance = createTransactionCommitter()
-      val context = new FakeCommitContext(recordTime = None)
-      val actual = instance.validateLedgerTime(context, aTransactionEntrySummary)
+  "validateLedgerTime" can {
+    "when the record time is not available" should {
+      "continue" in {
+        val result = instance.validateLedgerTime(
+          contextWithTimeModelAndEmptyCommandDeduplication,
+          aTransactionEntrySummary)
 
-      actual match {
-        case StepContinue(_) => succeed
-        case StepStop(_) => fail
+        result match {
+          case StepContinue(_) => succeed
+          case StepStop(_) => fail
+        }
+      }
+
+      "compute and correctly set the minimum ledger time without deduplicateUntil" in {
+        instance.validateLedgerTime(
```

Comment:
Please merge these two test cases. Also, please explicitly spell out how you arrive at the expected values as they are non-obvious.